stock_available_mrp/migrations/17.0.1.0.0/pre-migration.py

```diff
@@ -1,10 +1,13 @@
 def migrate(cr, version):
+    # Requête pour vérifier l'existence des clés spécifiques dans la table `ir_config_parameter`
     cr.execute(
-        """SELECT id FROM ir_config_parameter
-        WHERE key = 'stock_available_mrp.stock_available_mrp_based_on'"""
+        """SELECT id, key FROM ir_config_parameter
+        WHERE key IN ('stock_available.stock_available_mrp_based_on', 'stock_available_mrp_based_on')"""
```

Comment:
```suggestion
        WHERE key IN ('stock_available_mrp.stock_available_mrp_based_on', 'stock_available_mrp_based_on')"""
```

Reply:
Hey @bizzappdev 

I could be wrong, I haven't yet invested in the script migration part of the OCA modules.

The "pre-migration" file is not supposed to be executed before the installation of our v17.0 module in which we have now defined the key "stock_available_mrp.stock_available_mrp_based_on"?
If yes, in this case in v16.0 the modules do not install a key (from what I observed on the OCA runbot) and if this would have been the case it is prefixed by the name of the module which adds it and previously this icp was introduced by the "stock_available" module

Do you have a v16.0 with this ICP to observe this?

Reply:
IMO you need to change the name of the existing key to v16.0 when migrating to v17.0
But I asked @pedrobaeza for advice (here https://github.com/OCA/stock-logistics-availability/pull/20#discussion_r1662103618 )

Reply:
I wouldn't change the key name for avoiding this kind of issues. It's still enough descriptive with the old name.

Reply:
I think out loud because this is my first participation in a migration with data changing from one module to another, and I am junior in this practice.

this would mean that:

- either the insertion of data in v17.0 should be "stock_available.stock_available_mrp_based_on" and not "stock_available_mrp.stock_available_mrp_based_on"
but I think there would be other side effects
https://github.com/BizzAppDev-Systems/stock-logistics-availability/blob/17.0-mig-stock_available_mrp-BAD/stock_available_mrp/data/ir_config_parameter.xml

- either here the recovery of the ICP should be based on external id "default_stock_available_mrp_based_on"
https://github.com/BizzAppDev-Systems/stock-logistics-availability/pull/1/commits/e7d3ae88dcbbb1d6adb8a073af8259524c4fc12a#diff-f9fbb7b16967da65f5b36aa479109a0fd7d7d423e81fe38f402 7870d4adb3187

Reply:
True! The XML-ID should change for proper uninstallation, and that requires a migration script, so on such migration script, you can also rename the key if you prefer.